internal/cmd/beta/network-interface/update/update.go

```diff
@@ -0,0 +1,242 @@
+	var allowedAddressesInner []iaas.AllowedAddressesInner
+	if allowedAddresses != nil && len(*allowedAddresses) > 0 {
+		allowedAddressesInner = make([]iaas.AllowedAddressesInner, len(*allowedAddresses))
+		for k, v := range *allowedAddresses {
+			// linter issue
+			value := v
+			allowedAddressesInner[k].String = &value
+		}
+	}
+
+	// check name length <= 63 and regex must apply
+	name := flags.FlagToStringPointer(p, cmd, nameFlag)
+	if name != nil {
+		if len(*name) > 63 {
+			return nil, &errors.FlagValidationError{
+				Flag:    nameFlag,
+				Details: fmt.Sprintf("name %s is too long (maximum length is 63 characters)", *name),
+			}
+		}
+		nameRegex := regexp.MustCompile(nameRegex)
+		if !nameRegex.MatchString(*name) {
+			return nil, &errors.FlagValidationError{
+				Flag:    nameFlag,
+				Details: fmt.Sprintf("name %s didn't match the required regex expression %s", *name, nameRegex),
+			}
+		}
+	}
+
+	// check security groups size and regex
+	securityGroups := flags.FlagToStringSlicePointer(p, cmd, securityGroupsFlag)
+	if securityGroups != nil && len(*securityGroups) > 0 {
+		securityGroupsRegex := regexp.MustCompile(securityGroupsRegex)
+		// iterate over them
+		for _, value := range *securityGroups {
+			if len(value) != 36 {
+				return nil, &errors.FlagValidationError{
+					Flag:    securityGroupsFlag,
+					Details: fmt.Sprintf("security groups uuid %s does not match (must be 36 characters long)", value),
+				}
+			}
+			if !securityGroupsRegex.MatchString(value) {
+				return nil, &errors.FlagValidationError{
+					Flag:    securityGroupsFlag,
+					Details: fmt.Sprintf("security groups uuid %s didn't match the required regex expression %s", value, securityGroupsRegex),
+				}
+			}
+		}
+	}
+
+	model := inputModel{
+		GlobalFlagModel: globalFlags,
+		NicId:           nicId,
+		NetworkId:       flags.FlagToStringPointer(p, cmd, networkIdFlag),
+		Labels:          flags.FlagToStringToStringPointer(p, cmd, labelFlag),
+		Name:            name,
+		NicSecurity:     flags.FlagToBoolPointer(p, cmd, nicSecurityFlag),
+		SecurityGroups:  securityGroups,
+	}
+
+	if allowedAddresses != nil {
+		model.AllowedAddresses = utils.Ptr(allowedAddressesInner)
+	}
+
+	if p.IsVerbosityDebug() {
+		modelStr, err := print.BuildDebugStrFromInputModel(model)
+		if err != nil {
+			p.Debug(print.ErrorLevel, "convert model to string for debugging: %v", err)
+		} else {
+			p.Debug(print.DebugLevel, "parsed input values: %s", modelStr)
+		}
+	}
+
+	return &model, nil
+}
+
+func buildRequest(ctx context.Context, model *inputModel, apiClient *iaas.APIClient) iaas.ApiUpdateNicRequest {
+	req := apiClient.UpdateNic(ctx, model.ProjectId, *model.NetworkId, model.NicId)
+
+	var labelsMap *map[string]interface{}
+	if model.Labels != nil && len(*model.Labels) > 0 {
+		// convert map[string]string to map[string]interface{}
+		labelsMap = utils.Ptr(map[string]interface{}{})
+		for k, v := range *model.Labels {
+			(*labelsMap)[k] = v
+		}
+	}
```

Comment:
```suggestion
	var labelsMap *map[string]interface{}
	if model.Labels != nil && len(*model.Labels) > 0 {
		// convert map[string]string to map[string]interface{}
        convertedMap := make(map[string]interface{}, len(*model.Labels))
        for k, v := range *model.Labels {
            convertedMap[k] = v
        }
        labelsMap = &convertedMap
	}
```

Reply:
Sorry, very minor gripe here, but you can directly make with the needed size and also avoid the ugly `(*x)` assignment